sdk/storage/azfile/directory/client.go

```diff
@@ -103,7 +104,7 @@ func NewClientFromConnectionString(connectionString string, shareName string, di
 		return nil, err
 	}
 
-	directoryPath = strings.ReplaceAll(directoryPath, "\\", "/")
+	directoryPath = filepath.ToSlash(directoryPath)
```

Comment:
This has introduced a regression and is one of the reasons for failures in CI. In this case, [the test](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/storage/azfile/directory/client_test.go#L197-L200) intentionally uses `\` chars for directory separators expecting them to be replaced but they aren't on Linux. I suspect the other failures are something similar.

Can you please revert these changes from the storage packages?